tested/languages/haskell/config.py

```diff
@@ -28,6 +29,10 @@
 
 
 class Haskell(Language):
+    def __init__(self, config: Optional["GlobalConfig"]):
+        super().__init__(config)
+        self.clean_types_regex = re.compile(r"\(([^:\s]*)\s*::\s*([A-Z][a-zA-Z0-9]*)\)")
```

Comment:
I think this should be a constant instead of a variable,

So `CLEAN_TYPES_REGEX =`

Also is this regex completly save? Eg is it possible that it also matches a type expression within a string?

If typing is optional, it would probably be better not to add the type declarations in 'generators.py', instead of adding them there, running the tests with typing and then obfuscating those types to the end user.